pkg/process/checks/process.go

```diff
@@ -133,7 +151,15 @@ func (p *ProcessCheck) Run(cfg *config.AgentConfig, groupID int32) ([]model.Mess
 		p.lastCtrRates = util.ExtractContainerRateMetric(ctrList)
 		p.lastCtrIDForPID = ctrByProc
 		p.lastRun = time.Now()
-		return nil, nil
+		p.storeCreateTimes()
+
+		if collectRealTime {
+			p.realtimeLastCPUTime = p.lastCPUTime
+			p.realtimeLastProcs = procsToStats(p.lastProcs)
+			p.realtimeLastCtrRates = p.lastCtrRates
+			p.realtimeLastRun = p.lastRun
```

Comment:
nit: Would be nice if the storing of the last checks was in a function, but maybe in a future PR.

Reply:
Agreed, I will do some refactoring in a separate PR along with more tests, now that the probe is a mockable interface it should be fairly easy to do that.

Reply:
Even better we could wrap the `last***` and `realtimeLast***` variables into structs.

Reply:
yep, that's a great idea and it would prevent obvious typos in some cases